### ISSUE TITLE
Web - Refactor callbacks using promises

### ISSUE THREAD

**Issue by dvstans** [ORNL/DataFed#86]
The web portal javascript is becoming difficult to maintain due to a proliferation of complex callback-based asynchronous functions. This can be improved by refactoring to use promises.

**Comment by dvstans** [comment]
This may not be worthwhile if we plan on replacing the front end in the next version of DataFed.

**Comment by JoshuaSBrown** [comment]
Yes, I would hold off on this, if we ever get a Web Developer, using a framework like React might negate a lot of this.
